src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/QueueCapacity.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.CPU;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.NETWORK;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ThreadPoolEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class QueueCapacity implements Action {
+
+  public static final String NAME = "queue_capacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
+
+  private int currentCapacity;
+  private int desiredCapacity;
+  private ThreadPoolEnum threadPool;
+  private NodeKey esNode;
+
+  private Map<ThreadPoolEnum, Integer> lowerBound = new HashMap<>();
+  private Map<ThreadPoolEnum, Integer> upperBound = new HashMap<>();
+
+  public QueueCapacity(NodeKey esNode, ThreadPoolEnum threadPool, int currentCapacity, boolean increase) {
+    setBounds();
+    int STEP_SIZE = 50;
+    this.esNode = esNode;
+    this.threadPool = threadPool;
+    this.currentCapacity = currentCapacity;
+    int desiredCapacity = increase ? currentCapacity + STEP_SIZE : currentCapacity - STEP_SIZE;
+    setDesiredCapacity(desiredCapacity);
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public boolean isActionable() {
+    return desiredCapacity != currentCapacity;
+  }
+
+  @Override
+  public int coolOffPeriodInSeconds() {
+    return COOL_OFF_PERIOD_IN_SECONDS;
+  }
+
+  @Override
+  public List<NodeKey> impactedNodes() {
+    return Collections.singletonList(esNode);
+  }
+
+  @Override
+  public Map<NodeKey, ImpactVector> impact() {
+    ImpactVector impactVector = new ImpactVector();
+    if (desiredCapacity > currentCapacity) {
+      impactVector.increasesPressure(HEAP, CPU, NETWORK);
```

Comment:
Is it true that increasing the queue capacity will increase the CPU load ? You mean for CPU bound processing of queue items ?

Reply:
Since it increases throughput, I think it would increase the CPU load. 
CPU ends up doing more amortized work per unit time. Bursty requests that would've otherwise been rejected and retried, are now accepted by the larger queue and processed; making cpu less idle and increasing throughput.

Reply:
Sorry about that, my point was because this is a PR for the framework, I am suggesting that what the decider predicts about resource usage might be taken as a hint and not as a definitive statement. 
But I guess, this can be a future improvement for the learning system.